test/node-report/helper.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const fs = require('fs');
+
+const REPORT_SECTIONS = [ 'header',
+                          'javascript_stack',
+                          'native_stack',
+                          'javascript_heap',
+                          'resource_usage',
+                          'libuv',
+                          'environment_variables',
+                          'user_limits',
+                          'shared_objects' ];
+
+exports.findReports = (pid) => {
+  // Default filenames are of the form
+  // node-report.<date>.<time>.<pid>.<seq>.json
+  const format = '^node-report\\.\\d+\\.\\d+\\.' + pid + '\\.\\d+\\.json$';
+  const filePattern = new RegExp(format);
+  const files = fs.readdirSync('.');
+  return files.filter((file) => filePattern.test(file));
+};
+
+exports.validate = (report, options) => {
+  fs.readFile(report, (err, data) => {
+    this.validateContent(data, options);
+  });
+};
+
+exports.validateContent = function validateContent(data, options) {
+  const report = JSON.parse(data);
+  const comp = Object.keys(report);
+
+  // Check all sections are present
+  REPORT_SECTIONS.forEach((section) => {
+    assert.ok(comp.includes(section));
+  });
+
+/*  process.on('exit', () => {
+    const format = '^node-report\\.\\d+\\.\\d+\\.\\d+\\.\\d+\\.json$';
+    const filePattern = new RegExp(format);
+    const files = fs.readdirSync('.');
+    const file = files.filter((f) => filePattern.test(f));
+    file.forEach((f) => {
+      try {
+        fs.unlinkSync(f);
+      } catch (e) {}
+    });
+  }); */
```

Comment:
Unused?

Reply:
this will need to be re-instated; commented for testing purpose. the generated report files need to be cleaned up upon the test completion, or else they occupy space. (I don't know how CI systems manage their spaces)

Reply:
this code will go off when we run tests in parallel and in temporary folders that should take care of removal of test-generated files.

Reply:
this code is now removed; as we run the tests on temp folder which is manages by the test harness.